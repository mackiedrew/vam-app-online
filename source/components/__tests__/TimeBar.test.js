import Subject, { Slice } from "../TimeBar";

const mockProps = {
  view: {
    start: 0,
    end: 44100 * 10 // 10 seconds
  }
};

describe("<TimeBar />", () => {
  it("renders without crashing", () => {
    shallow(<Subject {...mockProps} />);
  });
});

describe("<Slice />", () => {
  it("renders without crashing", () => {
    shallow(<Slice sample={400} sampleSpan={441000} />);
  });

  it("renders without when sample span over 60 seconds", () => {
    shallow(<Slice sample={400} sampleSpan={44100 * 60} />);
  });

  it("renders without when sample span over 60 minutes", () => {
    shallow(<Slice sample={400} sampleSpan={44100 * 60 * 61} />);
  });
});
