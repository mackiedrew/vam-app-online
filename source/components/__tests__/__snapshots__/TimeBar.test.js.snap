// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TimeBar /> renders correctly with props 1`] = `
<div
  className="time-bar"
>
  <div
    className="slice"
  >
    00s:000
  </div>
  <div
    className="slice"
  >
    01s:000
  </div>
  <div
    className="slice"
  >
    02s:000
  </div>
  <div
    className="slice"
  >
    03s:000
  </div>
  <div
    className="slice"
  >
    04s:000
  </div>
  <div
    className="slice"
  >
    05s:000
  </div>
  <div
    className="slice"
  >
    06s:000
  </div>
  <div
    className="slice"
  >
    07s:000
  </div>
  <div
    className="slice"
  >
    08s:000
  </div>
  <div
    className="slice"
  >
    09s:000
  </div>
</div>
`;
